samples/snippets/src/main/java/pubsub/SubscribeWithExactlyOnceConsumerWithResponseExample.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_subscriber_exactly_once]
+
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class SubscribeWithExactlyOnceConsumerWithResponseExample {
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String topicId = "your-topic-id";
+    String subscriptionId = "your-subscription-id";
+
+    subscribeWithExactlyOnceConsumerWithResponseExample(projectId, topicId, subscriptionId);
+  }
+
+  public static void subscribeWithExactlyOnceConsumerWithResponseExample(
+      String projectId, String topicId, String subscriptionId) {
+    ProjectSubscriptionName subscriptionName =
+        ProjectSubscriptionName.of(projectId, subscriptionId);
+
+    // Instantiate an asynchronous message receiver using `AckReplyConsumerWithResponse`
+    // instead of `AckReplyConsumer` to get a future that tracks the result of the ack call.
+    // When exactly once delivery is enabled on the subscription, the message is guaranteed
+    // to not be delivered again if the ack future succeeds.
+    MessageReceiverWithAckResponse receiverWithResponse =
+        (PubsubMessage message, AckReplyConsumerWithResponse consumerWithResponse) -> {
+          // Handle incoming message, then ack the received message, and receive the response.
+          Future<AckResponse> ackResponseFuture = consumerWithResponse.ack();
+
+          try {
+            // Retrieve the completed future.
+            AckResponse ackResponse = ackResponseFuture.get();
+            
+            // For subscriptions with exactly once enabled, the AckResponse will
+            // return success OR permanent failures.
+            // For subscriptions without exactly once enabled the AckResponse will
+            // return success for messages ack/nack'd OR permanent failures.
```

Comment:
```suggestion
```

The wording somehow gives me the impression that in the first case of exactly once enabled, the AckResponse will return success for messages ack'ed OR permanent failures.  I'm not sure what's there to gain with this comment.